cmd/spire-agent/cli/command/api_fetch.go

```diff
@@ -0,0 +1,260 @@
+package command
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/spiffe/spire/proto/api/workload"
+
+	"google.golang.org/grpc"
+)
+
+type APIFetchConfig struct {
+	silent     bool
+	socketPath string
+	spiffeID   string
+	writePath  string
+}
+
+type APIFetch struct {
+	config *APIFetchConfig
+}
+
+func (APIFetch) Synopsis() string {
+	return "Fetches SVIDs from the Workload API"
+}
+
+func (f APIFetch) Help() string {
+	err := f.parseConfig([]string{"-h"})
+	return err.Error()
+}
+
+func (f *APIFetch) Run(args []string) int {
+	err := f.parseConfig(args)
+	if err != nil {
+		fmt.Println(err.Error())
+		return 1
+	}
+
+	client, err := f.dial()
+	if err != nil {
+		fmt.Println(err.Error())
+		return 1
+	}
+
+	start := time.Now()
+	bundles, err := f.fetchBundles(context.TODO(), client)
+	respTime := time.Now().Sub(start)
+	if err != nil {
+		fmt.Println(err.Error())
+		return 1
+	}
+
+	if !f.config.silent {
+		f.printBundles(bundles, respTime)
+	}
+
+	if f.config.writePath != "" {
+		err = f.writeBundles(bundles)
+		if err != nil {
+			fmt.Println(err.Error())
+			return 1
+		}
+	}
+
+	return 0
+}
+
+func (f *APIFetch) parseConfig(args []string) error {
+	fs := flag.NewFlagSet("fetch", flag.ContinueOnError)
+	c := &APIFetchConfig{}
+	fs.BoolVar(&c.silent, "silent", false, "Suppress stdout")
+	fs.StringVar(&c.spiffeID, "spiffeID", "", "Retrieve only a specific SPIFFE ID (optional)")
+	fs.StringVar(&c.socketPath, "socketPath", "/tmp/agent.sock", "Path to the Workload API socket")
+	fs.StringVar(&c.writePath, "write", "", "Write SVID data to the specified path (optional)")
+
+	f.config = c
+	return fs.Parse(args)
+}
+
+func (f *APIFetch) fetchBundles(ctx context.Context, c workload.WorkloadClient) (*workload.Bundles, error) {
+	var resp *workload.Bundles
+	var err error
+
+	if f.config.spiffeID == "" {
+		resp, err = c.FetchAllBundles(ctx, &workload.Empty{})
+	} else {
+		id := &workload.SpiffeID{Id: f.config.spiffeID}
+		resp, err = c.FetchBundles(ctx, id)
+	}
+
+	if err != nil {
+		return &workload.Bundles{}, err
+	}
+
+	return resp, nil
+}
+
+func (f APIFetch) printBundles(bundles *workload.Bundles, respTime time.Duration) {
+	lenMsg := fmt.Sprintf("Fetched %v bundle", len(bundles.Bundles))
+	if len(bundles.Bundles) != 1 {
```

Comment:
if len(bundles.Bundles) > 1

Reply:
Oddly, plural applies to 0: `Fetched 0 bundles` vs `Fetched 0 bundle`